pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -465,6 +465,10 @@ func (k *Kubernetes) Transform(komposeObject kobject.KomposeObject, opt kobject.
 
 		// Generate pod only and nothing more
 		if service.Restart == "no" || service.Restart == "on-failure" {
+			// Error out if Controller Object is specified with restart: 'on-failure'
+			if opt.IsDeploymentFlag || opt.IsDaemonSetFlag || opt.IsReplicationControllerFlag {
+				logrus.Fatalf("Controller object cannot be specified with restart: 'on-failure'")
+			}
```

Comment:
Don't really understand this part. If a user __doesn't__ pass in these flags, it won't error out?